VideoLocker/src/main/java/org/edx/mobile/view/WebViewFindCoursesActivity.java

```diff
@@ -35,21 +42,33 @@ protected void onCreate(Bundle savedInstanceState) {
 
         webView = (WebView) findViewById(R.id.webview);
         webView.loadUrl(environment.getConfig().getCourseDiscoveryConfig().getCourseSearchUrl());
+
+        DrawerLayout mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerToggle = new ActionBarDrawerToggle(this, mDrawerLayout,
+                R.string.label_open_drawer, R.string.label_close_drawer) {
+            @Override
+            public void onDrawerSlide(View drawerView, float slideOffset) {
+                super.onDrawerSlide(drawerView, slideOffset);
+                invalidateOptionsMenu();
```

Comment:
Hmm, this seems to be a pretty expensive way to do things, cause it'll be called constantly during the sliding of the drawer, causing the menu to be constantly invalidated and reinflated as well. A better option might be to override the `onDrawerStateChanged()` method, e.g.:

```` java
@Override
public void onDrawerStateChanged(int newState) {
    if (newState == DrawerLayout.STATE_SETTLING) {
        // Note that isDrawerOpen() will return the old state that's being changed
        searchView.setVisibility(!drawerLayout.isDrawerOpen(GravityCompat.START) ?
                View.GONE : View.VISIBLE);
    }
}
````